google/cloud/storage/internal/curl_client.cc

```diff
@@ -201,6 +228,13 @@ std::pair<Status, IamPolicy> CurlClient::SetBucketIamPolicy(
 
 std::pair<Status, ObjectMetadata> CurlClient::InsertObjectMedia(
     InsertObjectMediaRequest const& request) {
+  if (not request.HasOption<IfMetagenerationNotMatch>() and
+      not request.HasOption<IfGenerationNotMatch>() and
+      not request.HasOption<QuotaUser>() and
+      not request.HasOption<Projection>() and request.HasOption<Fields>() and
+      request.GetOption<Fields>().value().empty()) {
+    return InsertObjectMediaXml(request);
```

Comment:
The response will be different when using XML API versus JSON API. Is this considered?

Reply:
Yes, this is only used when `Fields("")` is one of the parameters, and in that case both the XML API and the JSON API return the same thing (an empty metadata object).

Reply:
Ahh, so a user has to specifically set Fields("") (I don't want any metadata) to go through this path. 

Sg, thanks!